### PR TITLE
libpaho-mqtt 1.3.12 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- needed for #130695
- debian inclusion, https://packages.debian.org/sid/libpaho-mqtt-dev

### CODE REVIEW

**Comment by chenrui333** [comment]
not sure if we want to go with `libpaho-mqtt3` or not

**Comment by carlocab** [comment]
> not sure if we want to go with `libpaho-mqtt3` or not

How do other package managers call this?

File: Formula/libpaho-mqtt.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  desc "Eclipse Paho C client library for MQTT"
+  homepage "https://eclipse.github.io/paho.mqtt.c/"
+  url "https://github.com/eclipse/paho.mqtt.c/archive/refs/tags/v1.3.12.tar.gz"
+  sha256 "6a70a664ed3bbcc1eafdc45a5dc11f3ad70c9bac12a54c2f8cef15c0e7d0a93b"
+  license "EPL-2.0"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```
Comment (carlocab) [inline Formula/libpaho-mqtt.rb:11]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", "-DBUILD_SHARED_LIBS=ON", *std_cmake_args
```
Unless it already builds shared libraries, but it doesn't look like it does by default.

Comment (chenrui333) [reply Formula/libpaho-mqtt.rb:11]: it does build shared libs by default

```
$ tree /opt/homebrew/Cellar/libpaho-mqtt/1.3.12/
/opt/homebrew/Cellar/libpaho-mqtt/1.3.12/
├── INSTALL_RECEIPT.json
├── LICENSE
├── README.md
├── about.html
├── bin
│   └── MQTTVersion
├── include
│   ├── MQTTAsync.h
│   ├── MQTTClient.h
│   ├── MQTTClientPersistence.h
│   ├── MQTTExportDeclarations.h
│   ├── MQTTProperties.h
│   ├── MQTTReasonCodes.h
│   └── MQTTSubscribeOpts.h
├── lib
│   ├── cmake
│   │   └── eclipse-paho-mqtt-c
│   │       ├── eclipse-paho-mqtt-cConfig-release.cmake
│   │       ├── eclipse-paho-mqtt-cConfig.cmake
│   │       └── eclipse-paho-mqtt-cConfigVersion.cmake
│   ├── libpaho-mqtt3a.1.3.12.dylib
│   ├── libpaho-mqtt3a.1.dylib -> libpaho-mqtt3a.1.3.12.dylib
│   ├── libpaho-mqtt3a.dylib -> libpaho-mqtt3a.1.dylib
│   ├── libpaho-mqtt3c.1.3.12.dylib
│   ├── libpaho-mqtt3c.1.dylib -> libpaho-mqtt3c.1.3.12.dylib
│   └── libpaho-mqtt3c.dylib -> libpaho-mqtt3c.1.dylib
├── notice.html
└── share
    └── doc
        └── Eclipse Paho C
            ├── CONTRIBUTING.md
            ├── README.md
            ├── edl-v10
            ├── epl-v20
            ├── notice.html
            └── samples
                ├── MQTTAsync_publish.c
                ├── MQTTAsync_publish_time.c
                ├── MQTTAsync_subscribe.c
                ├── MQTTClient_publish.c
                ├── MQTTClient_publish_async.c
                ├── MQTTClient_subscribe.c
                ├── paho_c_pub.c
                ├── paho_c_sub.c
                ├── paho_cs_pub.c
                ├── paho_cs_sub.c
                └── pubsub_opts.c

10 directories, 38 files
```

But happy to add one.

**Comment by chenrui333** [comment]
> > not sure if we want to go with `libpaho-mqtt3` or not
> 
> How do other package managers call this?

not much consensus, this is the closest one in repology, https://repology.org/project/libpaho-mqtt3/versions

File: Formula/libpaho-mqtt.rb (hunk 1 above)
Comment (carlocab) [reply Formula/libpaho-mqtt.rb:11]: No need if it builds shared libraries by default.

**Review by carlocab — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**